src/code/z_camera.c

```diff
@@ -55,31 +256,184 @@ s32 Camera_IsSwimming(Camera* camera) {
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB8C8.s")
+s32 Camera_IsDiving(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->stateFlags2 & PLAYER_STATE2_800;
+    } else {
+        return 0;
+    }
+}
+
+s32 Camera_IsPlayerFormZora(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->transformation == PLAYER_FORM_ZORA;
+    } else {
+        return false;
+    }
+}
+
+s32 func_800CB924(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->stateFlags3 & PLAYER_STATE3_1000;
+    } else {
+        return 0;
+    }
+}
+
+s32 func_800CB950(Camera* camera) {
+    Player* player;
+    s32 phi_v0;
+    s32 ret;
+    f32 yDiff;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        yDiff = Camera_fabsf(camera->focalActorPosRot.pos.y - camera->playerFloorHeight);
+
+        phi_v0 = false;
+        if (yDiff < 11.0f) {
+            phi_v0 = true;
+        }
+
+        ret = phi_v0;
+
+        if (!ret) {
+
+            ret = false;
+
+            if (camera->focalActor->gravity > -0.1f) {
+                ret = true;
+            }
+
+            player = (Player*)camera->focalActor;
+            if (!ret) {
+                // Using zora fins
+                ret = player->stateFlags1 & PLAYER_STATE1_200000;
+                ret = !!ret;
```

Comment:
```suggestion
                ret = !!ret;
```
Not not, who's there? Okay, I got that out of my system....